### ISSUE TITLE
scroll-wheel setting persists

### ISSUE THREAD

**Issue by ricalvrod** [pwr-Solaar/Solaar#971]
I have an MX Master 3 mouse and after changing smart shift the setting became Persistent. Even if I reset it or update the mouse with de Logitech options app that setting never return to original. I want to reset it to original, but there is no option to do so. And i don't know anymore the original value. If someone can help i would be grateful. Also at leat my unit never display the battery percentage correctly, doesn't change from 100%.

**Comment by pfps** [comment]
Yes, Solaar does not keep track of the original setting.  To find the original setting you can stop Solaar, edit ~/.config/solaar/config.json to remove the setting in question (which may require a bit of guessing), turn off and on your mouse, and restart Solaar.  Solaar will then pick up the setting from the device and keep track of that.

Instead of removing the setting you could move config.json to another place temporarily and later move it back.

**Comment by pfps** [comment]
Also, the MX Master will report 100% until the battery is about half discharged.

**Comment by ricalvrod** [comment]
I tried everything but the problem persists. When I turn of the mouse the setting i changed (Snart Shift) stays, even if i change to other computer. That is really strange, because the original logitech options doesn't save any settings, but when changing in solaar stays persistent in the mouses memory.

**Comment by pfps** [comment]
The code for this Smart Shift sets not only the current value but also the default value.  This is not how other settings work, so I'm probably going to change it to work like other settings.  With this proposed change the mouse will go back to its default state when it is turned off and on again and Solaar is not running.  The problem with this proposed change is that the default will be the default that Solaar set, not the out-of-the-box default.  Comments are welcome on this point.

To answer your original question the OOB default is 16 according to Logitech documentation.

**Comment by ricalvrod** [comment]
Oh so that is really the problem. Yup it is a better idea to have that setting working as others. Oh!!! I see the problem, now after solaar changed the setting, even with the update, the default for everyone will be solaar setted. I think for most user they will not notice it but a suggestion is to add some type of persistency check box, when checked those values will be defaulted in the mouse. For people that uses also Bluetooth it becomes a feature that even the manufacture didn't have.

As the response for the original question, first thank you a lot but I am nor really sure that 16 is the value... When I set it to 16 in solaar it does not resemble to original.

**Comment by pfps** [comment]
Well 16 is the number in the Logitech documentation.  This is not to say that the documentation is correct nor that Logitech software doesn't change the number.  But I do seem to vaguely remember that the number that was shown for my MX Master 3 was more like 25 before I started fiddling with the setting.

**Comment by pfps** [comment]
On further thought maybe the OOB default is 0x16.

**Comment by ricalvrod** [comment]
I as trying to came up with the number using the Logitech options software. Because it has a reset to defaults button, but Logitech is just dumb and it resets the local program and not the mouse defaults. Nevertheless with the print from Logitech option and one from Solaar I came up with some sort of number using GIMP, and I think the number is like 8. But after try it I am really not sure. From what I remember because I started fiddling with this yesterday the number should be something more like between 10 and 14. But at this scale is now impossible to know. Unless someone buys a new MX Master 3 hehehe.

0x16 is really a good bet but I don't think is that much...

**Comment by pfps** [comment]
By the way, the easiest way to get this documentation is from
https://lekensteyn.nl/files/logitech/x2110_smartshift.html

**Comment by ricalvrod** [comment]
That is just amazing information. I don't know why that feature is not documented in the Logitech GitHub or google drive.

By the way I am almost 99% certain that 16 is not the value or not is 0x16. I did a little experiment and resulted in success. Whit the original Logitech app I reverted the settings and if you don't turn off the mouse (witch i did not) the reseated value stays and in sollar i could finally read it and it is **10**. I think i can finally rest XD. 

If someone has other information and for some reason it really isn't 10 pls let me know.
